### ISSUE TITLE
ci: v9 release pipelines should execute same checks like PR

### ISSUE THREAD

**Issue by Hotell** [microsoft/fluentui#24811]
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

We run test/lint/build in separate task synchronously for v9 release pipeline while on PR everything is run in parallel. This slows down release pipeline time.
 
**PR:**
 
<img width="847" alt="image" src="https://user-images.githubusercontent.com/1223799/190372836-5797795c-0d11-45c2-83d7-986235284fc3.png">

 
**V9 release:** 
 
<img width="675" alt="image" src="https://user-images.githubusercontent.com/1223799/190372889-92531672-c597-4c9c-8803-e0a925f2f929.png">

Last but not least we don't run type-checks, which might cause issue for release as we dont enforce PR branch always up to date so things may break. we should unify these pipelines.

### Have you discussed this feature with our team

_v-build_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
